client/components/upgrades/google-apps/dialog/users.jsx

```diff
@@ -46,75 +50,80 @@ var GoogleAppsUsers = React.createClass( {
-					onClick={ this.recordInputFocus.bind( this, index, 'Last Name' ) } />
-			</fieldset>
+			<div className="google-apps-dialog__user-fields" key={ `google-apps-dialog-user-${ index }` }>
+				<FormFieldset>
+					<FormTextInputWithAffixes
+						className={ this.fieldClasses( 'email' ) }
+						placeholder={ this.translate( 'e.g. %(example)s', { args: { example: contactText } } ) }
+						name="email"
+						value={ user.email.value }
+						suffix={ '@' + this.props.domain }
+						isError={ user.email.error }
+						onChange={ this.updateField.bind( this, index ) }
+						onBlur={ this.props.onBlur }
+						onClick={ this.recordInputFocus.bind( this, index, 'Email' ) } />
+					{ user.email.error ? <FormInputValidation text={ user.email.error } isError={ true } /> : null }
+				</FormFieldset>
+
+				<FormFieldset className={ this.fieldClasses( 'first-name' ) }>
+					<FormTextInput
+						placeholder={ this.translate( 'First Name' ) }
+						name="firstName"
+						value={ user.firstName.value }
+						maxLength={ 60 }
+						isError={ user.firstName.error }
+						onChange={ this.updateField.bind( this, index ) }
+						onBlur={ this.props.onBlur }
+						onClick={ this.recordInputFocus.bind( this, index, 'First Name' ) } />
+					{ user.firstName.error ? <FormInputValidation text={ user.firstName.error } isError={ true } /> : null }
+				</FormFieldset>
+
+				<FormFieldset className={ this.fieldClasses( 'last-name' ) }>
+					<FormTextInput
+						placeholder={ this.translate( 'Last Name' ) }
+						name="lastName"
+						value={ user.lastName.value }
+						maxLength={ 60 }
+						isError={ user.lastName.error }
+						onChange={ this.updateField.bind( this, index ) }
+						onBlur={ this.props.onBlur }
+						onClick={ this.recordInputFocus.bind( this, index, 'Last Name' ) } />
+					{ user.lastName.error ? <FormInputValidation text={ user.lastName.error } isError={ true } /> : null }
+				</FormFieldset>
+			</div>
 		);
 	},
 
-	recordInputFocus: function( index, fieldName ) {
-		var field = this.props.fields[ index ],
+	recordInputFocus( index, fieldName ) {
+		const field = this.props.fields[ index ],
 			inputValue = field ? field.value : '';
 
 		this.recordEvent( 'inputFocus', index, fieldName, inputValue );
 	},
 
-	addUser: function( event ) {
+	addUser( event ) {
 		event.preventDefault();
 
 		this.recordEvent( 'addUserClick', this.props.analyticsSection );
 
-		var updatedFields = this.props.fields.concat( [ this.getNewUserFields() ] );
+		const updatedFields = this.props.fields.concat( [ this.getNewUserFields() ] );
 		this.props.onChange( updatedFields );
 	},
 
-	updateField: function( index, fieldName, event ) {
-		var newValue, updatedFields;
+	updateField( index, event ) {
 		event.preventDefault();
 
-		newValue = event.target.value;
-
-		updatedFields = clone( this.props.fields );
+		const newValue = event.target.value,
+			fieldName = event.target.name,
+			updatedFields = clone( this.props.fields );
```

Comment:
Because I am changing the value of this variable, not its reference.
From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const):
> The const declaration creates a read-only reference to a value. It does not mean the value it holds is immutable, just that the variable identifier cannot be reassigned.

JS does not have a qualifier for what you want to achieve - immutability (unfortunately - oh, glorious C++, how I miss your mighty `const`). There's [immutable.js](https://facebook.github.io/immutable-js/), though :)